internal/racetime/api.go

```diff
@@ -69,19 +79,114 @@ type CategoryResponse struct {
 	CurrentRaces      []RaceData `json:"current_races"`
 }
 
+type LeaderboardsResponse struct {
+	Goal      string `json:"goal"`
+	NumRanked string `json:"num_ranked"`
+	Rankings  struct {
+		User         string `json:"user"`
+		Place        string `json:"place"`
+		PlaceOrdinal string `json:"place_ordinal"`
+		Score        int    `json:"score"`
+		TimesRaced   string `json:"times_raced"`
+	}
+}
+
+type UserSearchParameters struct {
+	Name string
+	Scrim string
+	Term string
+}
+
 // CategoryDetail fetches the race data of a specific racetime.gg category
 func CategoryDetail(c config.Racetime, category string) (*CategoryResponse, error) {
-	resp, err := http.Get(fmt.Sprintf("%s/%s/data", c.URL, category))
+	content, err := Get(c, fmt.Sprintf("%s/data", category), nil)
+
+	var cr *CategoryResponse
+	json.Unmarshal([]byte(content), cr)
 	if err != nil {
 		return nil, err
 	}
-	defer resp.Body.Close()
 
-	var cr *CategoryResponse
-	err = json.NewDecoder(resp.Body).Decode(cr)
+	return cr, nil
+}
+
+func CategoryLeaderboards(c config.Racetime, category string) (*LeaderboardsResponse, error) {
+	content, err := Get(c, fmt.Sprintf("%s/leaderboards/data", category), nil)
+
+	var cl *LeaderboardsResponse
+	json.Unmarshal([]byte(content), cl)
 	if err != nil {
 		return nil, err
 	}
 
-	return cr, nil
+	return cl, nil
 }
+
+// RaceDetail fetches the race data of a specific racetime.gg race
+func RaceDetail(c config.Racetime, category string, race string) (*RaceData, error) {
+	content, err := Get(c, fmt.Sprintf("%s/%s/data", category, race), nil)
+
+	var rd *RaceData
+	json.Unmarshal([]byte(content), rd)
+	if err != nil {
+		return nil, err
+	}
+
+	return rd, nil
+}
+
+func PastUserRaces(c config.Racetime, user string, showEntrants bool, page int){
+	content, err := Get(c, fmt.Sprintf("user/%s/races/data", user), nil)
+
+	var pur *RaceData
+	json.Unmarshal([]byte(content), pur)
+	if err != nil {
+	}
+}
+
+func UserSearch(c config.Racetime, user string){
+	parameters := UserSearchParameters{
+		Name: "colfra",
+	}
+
+	m := structs.Map(parameters)
+	content, _ := Get(c, "user/search", m)
+	var userSearchResults UserDataResponse
+	json.Unmarshal([]byte(content), &userSearchResults)
+	log.Println(userSearchResults.Results[0].ID)
+}
+
+func GetTest(){
```

Comment:
Go actually has really good test tools built in. Usually you'd make a test file (ie. `racetime_test.go` or `api_test.go`) next to the file you're going to test, then append `_test` to the package name just for that file (eg. `racetime_test`). You can then start writing tests (one place to look for examples is my `lexer_test.go`).

Go should have built-ins for mocking http and such too.